src/codegen/abi.js

```diff
@@ -411,37 +430,93 @@ module.exports = class AbiCodeGenerator {
           )
           .forEach(member => returnType.addMember(member))
 
+        //Create types for Tuple outputs
+        outputs.forEach((output, index) => {
+          if (output.get('type') === 'tuple') {
+            types = types.push(
+              this._generateTupleType(
+                output,
+                index,
+                tupleResultParentType,
+                'function',
+                this.abi.name,
+              ).classes,
+            )
+          }
+        })
+
         // Add the type to the types we'll create
         types = types.push(returnType)
 
         returnType = tsCodegen.namedType(returnType.name)
       } else {
-        returnType = tsCodegen.namedType(
-          typesCodegen.ascTypeForEthereum(
-            member
-              .get('outputs')
+        let type = outputs
+          .get(0)
+          .get('type')
+        if (type === 'tuple') {
+          // Add the Tuple type to the types we'll create
+          types = types.push(
+            this._generateTupleType(
+              outputs.get(0),
+              0,
+              tupleResultParentType,
+              'function',
+              this.abi.name,
+            ).classes,
+          )
+
+          returnType = this._generateTupleClassName(
```

Comment:
Is this going to be compatible with the above `_generateTupleType`? Should we maybe add the name of the type to the object returned by `_generateTupleType` and then use that?